torch/autograd/gradcheck.py

```diff
@@ -1034,47 +1015,46 @@ def gradcheck(
     Returns:
         True if all differences satisfy allclose condition
     """
-    def fail_test(msg):
-        if raise_exception:
-            raise RuntimeError(msg)
-        return False
-
-    tupled_inputs = _as_tuple(inputs)
+    # This is just a wrapper that handles the raise_exception logic
+    args = locals()
+    args.pop("raise_exception")
+    if not raise_exception:
+        try:
+            return gradcheck_helper(**args)
+        except RuntimeError:
```

Comment:
This simplification is really nice.
I think you can make it even better by defining a custom Error type here like `GradCheckError`. And only catch that error in this except. That way, you won't intercept any of the user errors